src/app/components/products/products.component.html

```diff
@@ -0,0 +1,22 @@
+<div class="container">
+    <div class="card-container">
+      <div class="card" *ngFor="let product of products" [routerLink]="['/product',product.id]">
+        <mat-card>
+          <mat-card-header>
+            <div class="img"><img mat-card-image src="{{ product.image }}" alt="{{ product.title }}"></div>
+          </mat-card-header>
+          <mat-card-content class="card-content">
+            <mat-card-title>{{ product.title }}</mat-card-title>
+            <mat-card-subtitle>{{ product.price }}</mat-card-subtitle>
+            <p class="description">{{ product.description }}</p>
+          </mat-card-content>
+          <mat-card-actions>
+            <div class="button-container">
+              <button mat-button class="cart-button"><a>View Product</a></button>
+            </div>
+          </mat-card-actions>
+        </mat-card>
```

Comment:
product details can itself be a seprate component. and can be reused multiple place such as cart preview